src/app/commands/cmd_sprite_size.cpp

```diff
@@ -319,15 +328,77 @@ bool SpriteSizeCommand::onEnabled(Context* context)
 
 void SpriteSizeCommand::onExecute(Context* context)
 {
+#ifdef ENABLE_UI
+  const bool ui = (params().ui() && context->isUIAvailable());
+#endif
   const ContextReader reader(context);
   const Sprite* sprite(reader.sprite());
-  int new_width = (m_width ? m_width: int(sprite->width()*m_scaleX));
-  int new_height = (m_height ? m_height: int(sprite->height()*m_scaleY));
-  ResizeMethod resize_method = m_resizeMethod;
+  auto& params = this->params();
+
+  double ratio = sprite->width() / double(sprite->height());
+  if (params.scale.isSet()) {
+    params.setWidth(int(sprite->width() * params.scale()));
+    params.setHeight(int(sprite->height() * params.scale()));
+    params.setScaleX(params.scale());
+    params.setScaleY(params.scale());
+  }
+  else if (params.lockRatio()) {
+    if (params.width.isSet()) {
+      params.setHeight(int(params.width() / ratio));
+      params.setScaleX(params.width() / double(sprite->width()));
+      params.setScaleY(params.scaleX());
+    }
+    else if (params.height.isSet()) {
+      params.setWidth(int(params.height() * ratio));
+      params.setScaleY(params.height() / double(sprite->height()));
+      params.setScaleX(params.scaleY());
+    }
+    else if (params.scaleX.isSet()) {
+      params.setWidth(int(params.scaleX() * sprite->width()));
+      params.setHeight(int(params.scaleX() * sprite->height()));
+      params.setScaleY(params.scaleX());
+    }
+    else if (params.scaleY.isSet()) {
+      params.setWidth(int(params.scaleY() * sprite->width()));
+      params.setHeight(int(params.scaleY() * sprite->height()));
+      params.setScaleX(params.scaleY());
+    }
+    else {
+      params.setWidth(sprite->width());
+      params.setHeight(sprite->height());
+      params.setScaleX(1.0);
+      params.setScaleY(1.0);
+    }
+  }
+  else {
+    if (params.width.isSet()) {
+      params.setScaleX(params.width() / double(sprite->width()) * 100.0);
+    }
+    else if (params.scaleX.isSet()) {
+      params.setWidth(int(params.scaleX() * sprite->width() / 100.0));
+    }
+    else {
+      params.setWidth(sprite->width());
+      params.setScaleX(1.0);
+    }
+    if (params.height.isSet()) {
+      params.setScaleY(params.height() / double(sprite->height()) * 100.0);
+    }
+    else if (params.scaleY.isSet()) {
+      params.setHeight(int(params.scaleY() * sprite->height() / 100.0));
+    }
+    else {
+      params.setHeight(sprite->height());
+      params.setScaleY(1.0);
+    }
+  }
+  int new_width = params.width();
+  int new_height = params.height();
+  ResizeMethod resize_method = params.method();
 
 #ifdef ENABLE_UI
-  if (m_useUI && context->isUIAvailable()) {
-    SpriteSizeWindow window(context, new_width, new_height);
+  if (ui && context->isUIAvailable()) {
```

Comment:
```suggestion
  if (ui) {
```
No need for `context->isUIAvailable()` because it's already considered when the `ui` variable is defined.

Reply:
Got it!